### PR TITLE
Fix First/Last aggregate functions intermediate type and support decimal

### PR DESCRIPTION
**Pull request by Yohahaha**
Align First/Last aggregate function's intermediate result with Spark.

Support decimal.

### CODE REVIEW

**Comment by Yohahaha** [comment]
CI log shows the UT related with this PR passed.
```
2023-05-09T04:05:24.8387588Z 190/214 Test #200: velox_functions_spark_aggregates_test ..........................   Passed   14.95 sec
```

@zhejiangxiaomai Could you check these UT?
```
2023-05-09T04:42:46.8637254Z The following tests FAILED:
2023-05-09T04:42:46.8637913Z 	205 - velox_hdfs_file_test (Failed)
2023-05-09T04:42:46.8638371Z 	209 - velox_exec_test (Failed)
```

**Comment by Yohahaha** [comment]
MultiFragmentTest.customPlanNodeWithExchangeClient

**Comment by zhejiangxiaomai** [comment]
Can you run `velox_exec_test --gtest_filter=MultiFragmentTest.customPlanNodeWithExchangeClient` locally ?

**Comment by Yohahaha** [comment]
Failed UT can pass in my local machine

**Comment by Yohahaha** [comment]
image

**Comment by Yohahaha** [comment]
Hi @rui-mo could you help take a look?

**Comment by rui-mo** [comment]
@Yohahaha Thank you. Can we open a PR in Gluten to test this one? In that Gluten PR, the fetched Velox branch should be changed to your branch.  Here is an example: https://github.com/oap-project/gluten/pull/1496.
We can also consider adding some unit tests in [VeloxAggregateFunctionsSuite](https://github.com/oap-project/gluten/blob/main/backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala) to test the newly supported aggregate functions.

**Comment by Yohahaha** [comment]
Hi @rui-mo 

Related PR in Gluten is https://github.com/oap-project/gluten/pull/1581.
Velox side's UT is velox/functions/sparksql/aggregates/tests/FirstAggregateTest.cpp and velox/functions/sparksql/aggregates/tests/LastAggregateTest.cpp, already passed.

**Comment by Yohahaha** [comment]
Hi @zhejiangxiaomai 

Could we skip velox_hdfs_file_test?

File: velox/functions/sparksql/aggregates/FirstLastAggregate.cpp [hunk 1]
```diff
@@ -131,8 +138,29 @@ class FirstAggregate : public FirstLastAggregateBase<numeric, TData> {
       bool /* mayPushdown */) override {
     DecodedVector decoded(*args[0], rows);
 
-    rows.applyToSelected(
-        [&](vector_size_t i) { updateValue(i, groups[i], decoded); });
+    rows.applyToSelected([&](vector_size_t i) {
+      updateValue(decoded.index(i), groups[i], decoded.base());
```
Comment (rui-mo) [inline velox/functions/sparksql/aggregates/FirstLastAggregate.cpp:142]: What is the specific reason for this change, from `decoded` to `decoded.base()`?

**Review by rui-mo — COMMENTED** [review]
LGTM generally.

File: velox/functions/sparksql/aggregates/FirstLastAggregate.cpp (hunk 1 above)
Comment (Yohahaha) [reply velox/functions/sparksql/aggregates/FirstLastAggregate.cpp:142]: See line 162, for more cleaner `updateValue` abstraction.

Comment (rui-mo) [reply velox/functions/sparksql/aggregates/FirstLastAggregate.cpp:142]: OK. It helps to use a generic function, right?

Comment (Yohahaha) [reply velox/functions/sparksql/aggregates/FirstLastAggregate.cpp:142]: Yes

**Pull request merged**